### PR TITLE
Add initial implementation and micro app

### PR DESCRIPTION
**Pull request by puneet-pdx**
- Adds initial implementation of Callout(without the shape and leader) which supports the use case of displaying callout at a location
- Add the content slot parameter to MapView
- Add MapViewScope
- Add Micro app that displays a callout when user single taps on it.

### CODE REVIEW

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/MapView.kt [hunk 1]
```diff
@@ -161,12 +163,14 @@ public fun MapView(
     onLongPress: ((LongPressEvent) -> Unit)? = null,
     onTwoPointerTap: ((TwoPointerTapEvent) -> Unit)? = null,
     onPan: ((PanChangeEvent) -> Unit)? = null,
-    onDrawStatusChanged: ((DrawStatus) -> Unit)? = null
+    onDrawStatusChanged: ((DrawStatus) -> Unit)? = null,
+    content: (@Composable MapViewScope.() -> Unit)? = null
```
Comment (shubham7109) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/MapView.kt:167]: Would this need to be defined as a slot-param like using a `calloutContent`? Or would that be added in a future issue?

**Review by shubham7109 — APPROVED** [review]
Looks good, basic implementation works well and should be perfect to plug-in changes to build a Callout. I would recommend adding an empty `README.md` for the microapp in this PR.

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/MapView.kt (hunk 1 above)
Comment (puneet-pdx) [reply toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/MapView.kt:167]: We are sticking with this for now. This parameter might evolve in the future, it is yet to be determined. Check this space https://devtopia.esri.com/runtime/kotlin/discussions/3976

**Review by sorenoid — CHANGES_REQUESTED** [review]

File: microapps/MapViewCalloutApp/app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            excludes += "/META-INF/{AL2.0,LGPL2.1}"
+        }
+    }
+}
+
+// context receivers are not experimental anymore, but AS thinks they are.
+//https://youtrack.jetbrains.com/issue/KTIJ-21063
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile::class).all {
+    kotlinOptions.freeCompilerArgs += "-Xcontext-receivers"
+}
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply microapps/MapViewCalloutApp/app/build.gradle.kts:80]: This is included in all the micro apps currently and also part of the template app. We should create an issue to clean it up across all the micro apps if it is not needed ?

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    var mapPoint by remember { mutableStateOf ( null as Point? ) }
+
+    MapView(
+        modifier = Modifier.fillMaxSize(),
+        arcGISMap = arcGISMap,
+        onSingleTapConfirmed = { mapPoint = it.mapPoint},
+        content = { mapPoint?.let { Callout(location = mapPoint!!) { Text("Hello, World!", color = Color.Green) } } },
+    )
+}
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt:59]: Would't this classify as code duplication ?

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                initialViewpoint = Viewpoint(
+                    latitude = 39.8,
+                    longitude = -98.6,
+                    scale = 10e7
+                )
+            }
+        )
+    }
+
+    var mapPoint by remember { mutableStateOf ( null as Point? ) }
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt:51]: With the viewmodel this is not relevant

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.geoviewcompose
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt:18]: We can consider it later??

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt [hunk 6]
```diff
@@ -35,15 +35,14 @@ fun MainScreen(viewModel: MapViewModel) {
...
         onSingleTapConfirmed = viewModel::setMapPoint,
-        content = {
-            viewModel.mapPoint.collectAsState().value?.let {
-                Callout(location = viewModel.mapPoint.collectAsState().value!!) {
-                    Text(
-                        "Hello, World!",
-                        color = Color.Green
-                    )
-                }
+    ) {
```
[the remark this answers is not in the document]
Comment (sorenoid) [reply microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt:38]: also I don't see the reason for the double tap handler to be set.

**Review by sorenoid — CHANGES_REQUESTED** [review]
looks good, but I think the content should be null if the mapPoint is null. A couple other things.

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/MainScreen.kt]: The double tap handler leaked through.. I have update this

File: settings.gradle.kts [hunk 7]
```diff
@@ -88,6 +88,8 @@ include(":floor-filter-app")
 project(":floor-filter-app").projectDir = File(rootDir, "microapps/FloorFilterApp/app")
 include(":geoview-compose")
 project(":geoview-compose").projectDir = File(rootDir, "toolkit/geoview-compose")
+include(":map-view-callout-app")
+project(":map-view-callout-app").projectDir = File(rootDir, "microapps/mapviewcalloutapp/app")
```
[the remark this answers is not in the document]
Comment (puneet-pdx) [reply settings.gradle.kts]: a bunch of other micro apps are named this way, which threw me off..

**Review by sorenoid — APPROVED** [review]
👍

**Comment by puneet-pdx** [comment]
Thanks @sorenoid @shubham7109. Merging..

**Pull request merged**